sklearn/ensemble/_hist_gradient_boosting/grower.py

```diff
@@ -486,6 +487,12 @@ def split_next(self):
                 self._compute_best_split_and_push(right_child_node)
             self.total_find_split_time += time() - tic
 
+            for child in [left_child_node, right_child_node]:
+                if child.is_leaf:
+                    del child.histograms
+        del node.histograms
```

Comment:
not sure about this one: `node` might be the sibling of another node which may leverage the subtraction trick.

Reply:
hm I guess we would have computed its histogram already so that's fine